cloud/ec2.go

```diff
@@ -408,6 +428,24 @@ func (m *ec2Manager) spawnOnDemandHost(ctx context.Context, h *host.Host, ec2Set
 	instance := reservation.Instances[0]
 	h.Id = *instance.InstanceId
 
+	if useElasticIP && h.IPAllocationID != "" {
+		// Associate the IP address that was allocated for this host. This is
+		// necessary for the host to be usable because otherwise, it has no IP
+		// address.
+		// Unfortunately, this step is prone to timing issues because EC2 only
+		// allows an IP address to be associated with the host after the
+		// EC2 instance reaches the "running" state. If AWS is slow at getting
+		// the host to "running", this could run out of attempts to associate
+		// the IP address and the host would end up unusable.
+		// TODO (DEVPROD-17136): consider making this step more resilient
+		// against AWS timing problems.
+		grip.Error(message.WrapError(associateIPAddressForHost(ctx, m.client, h), message.Fields{
+			"message":       "host was created and allocated an elastic IP address but could not associate the host with the IP address; host will not be usable",
+			"host_id":       h.Id,
+			"allocation_id": h.IPAllocationID,
+		}))
```

Comment:
Why do we not want to return an error in this case?

Reply:
If we return an error, the host won't be inserted into the DB, which would result in a leaked host. I guess we could try terminating the host here and returning an error, but that would also be kind of finicky. I planned to make a separate job in DEVPROD-17136 to handle associating the address in a retry loop in case this fails. This call is more like a best-effort attempt to associate it immediately and won't actually guarantee the association succeeds.